examples/multi-pointer.c

```diff
@@ -0,0 +1,220 @@
+#define _POSIX_C_SOURCE 199309L
+#define _XOPEN_SOURCE 500
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <string.h>
+#include <unistd.h>
+#include <math.h>
+#include <assert.h>
+#include <wayland-server.h>
+#include <wayland-server-protocol.h>
+#include <xkbcommon/xkbcommon.h>
+#include <GLES2/gl2.h>
+#include <wlr/render/matrix.h>
+#include <wlr/render/gles2.h>
+#include <wlr/render.h>
+#include <wlr/backend.h>
+#include <wlr/backend/session.h>
+#include <wlr/types/wlr_keyboard.h>
+#include <wlr/types/wlr_output_layout.h>
+#include <wlr/xcursor.h>
+#include <wlr/types/wlr_cursor.h>
+#include <wlr/util/log.h>
+#include <wlr/types/wlr_list.h>
+#include "shared.h"
+#include "config.h"
+#include "cat.h"
+
+struct sample_state;
+
+struct sample_cursor {
+	struct sample_state *state;
+	struct wlr_input_device *device;
+	struct wlr_cursor *cursor;
+	struct wl_list link;
+
+	struct wl_listener cursor_motion;
+	struct wl_listener cursor_motion_absolute;
+	struct wl_listener cursor_button;
+	struct wl_listener cursor_axis;
+};
+
+struct sample_state {
+	struct compositor_state *compositor;
+	struct example_config *config;
+	struct wlr_xcursor *xcursor;
+	float default_color[4];
+	float clear_color[4];
+	struct wlr_output_layout *layout;
+	struct wl_list cursors; // sample_cursor::link
+};
+
+static void handle_output_frame(struct output_state *output,
+		struct timespec *ts) {
+	struct compositor_state *state = output->compositor;
+	struct sample_state *sample = state->data;
+	struct wlr_output *wlr_output = output->output;
+
+	wlr_output_make_current(wlr_output);
+
+	glClearColor(sample->clear_color[0], sample->clear_color[1],
+		sample->clear_color[2], sample->clear_color[3]);
+	glClear(GL_COLOR_BUFFER_BIT);
+
+	wlr_output_swap_buffers(wlr_output);
+}
+
+static void handle_output_add(struct output_state *ostate) {
+	struct sample_state *sample = ostate->compositor->data;
+
+	struct output_config *o_config =
+		example_config_get_output(sample->config, ostate->output);
+
+	if (o_config) {
+		wlr_output_transform(ostate->output, o_config->transform);
+		wlr_output_layout_add(sample->layout, ostate->output, o_config->x,
+			o_config->y);
+	} else {
+		wlr_output_layout_add_auto(sample->layout, ostate->output);
+	}
+
+	struct sample_cursor *cursor;
+	wl_list_for_each(cursor, &sample->cursors, link) {
+		example_config_configure_cursor(sample->config, cursor->cursor,
+			sample->compositor);
+
+		// TODO the cursor must be set depending on which surface it is
+		// displayed over which should happen in the compositor.
```

Comment:
Should we add a way to disable client cursors for compositors that don't want to use them? Could it be possible for compositors to have a single cursor theme?

Reply:
Just ignore the `set_cursor` request from clients.

Reply:
Yeah I think this todo is kind of old then before we knew about this stuff.

Reply:
Exactly, I should remove it (and the other ones across all examples).